### PR TITLE
Handle Karma exit codes in gulp

### LINKED ISSUE #9764: `gulp test` doesn't properly handle karma failures
When a karma (integration) test fails, `gulp test` returns a format error because it expects an `Error` object, while karma returns an exit code.

We must call `done()` after converting the exit code to an `Error`.

For example, see https://travis-ci.org/ampproject/amphtml/jobs/240195458

>    `at formatError (/home/travis/build/ampproject/amphtml/node_modules/gulp/bin/gulp.js:169:10)
    at Gulp.<anonymous> (/home/travis/build/ampproject/amphtml/node_modules/gulp/bin/gulp.js:195:15)
    at emitOne (events.js:77:13)
    at Gulp.emit (events.js:169:7)
    at Gulp.Orchestrator._emitTaskDone (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/index.js:264:8)
    at /home/travis/build/ampproject/amphtml/node_modules/orchestrator/index.js:275:23
    at finish (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/lib/runTask.js:21:8)
    at cb (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/lib/runTask.js:29:3)
    at /home/travis/build/ampproject/amphtml/build-system/tasks/runtime-test.js:206:5
    at removeAllListeners [as _onTimeout] (/home/travis/build/ampproject/amphtml/node_modules/karma/lib/server.js:380:7)
    at Timer.listOnTimeout (timers.js:92:15)`

### PR DESCRIPTION
**Pull request by rsimha-amp**
Fixes #9764

### CODE REVIEW

File: build-system/tasks/runtime-test.js [hunk 1]
```diff
@@ -203,7 +203,12 @@ gulp.task('test', 'Runs tests', argv.nobuild ? [] : ['build'], function(done) {
     util.log(util.colors.yellow(
         'Shutting down test responses server on localhost:31862'));
     server.emit('kill');
-    done(exitCode);
+    var error = exitCode;
+    if (exitCode && typeof exitCode !== 'Error') {
+      error = new Error('Karma test failed (error code: ' + exitCode + ')');
+      error.showStack = false;
+    }
+    done(error);
```
Comment (choumx) [inline build-system/tasks/runtime-test.js]: Nit: More readable if `error` wasn't multi-value, e.g.

```
if (exitCode && typeof exitCode !== 'Error') {
  var error = // ...
  done(error);
} else {
  done(exitCode);
}
```

Comment (rsimha-amp) [inline build-system/tasks/runtime-test.js]: done().

(heh!)

File: build-system/tasks/runtime-test.js [hunk 2]
```diff
@@ -203,7 +203,13 @@ gulp.task('test', 'Runs tests', argv.nobuild ? [] : ['build'], function(done) {
     util.log(util.colors.yellow(
         'Shutting down test responses server on localhost:31862'));
     server.emit('kill');
-    done(exitCode);
+    if (exitCode && typeof exitCode !== 'Error') {
```
Comment (erwinmombay) [inline build-system/tasks/runtime-test.js]: `typeof exitCode` works?

Comment (rsimha-amp) [inline build-system/tasks/runtime-test.js]: Yes, it works. Compare https://travis-ci.org/ampproject/amphtml/jobs/240439715 (with this change) against https://travis-ci.org/ampproject/amphtml/jobs/240195458 (without this change).

I wonder if it makes sense to set showStack to true. I figured I'd test it locally to see if it generates anything useful.

Comment (rsimha-amp) [inline build-system/tasks/runtime-test.js]: Just tried it, there's nothing in the stack. Karma merely returns an error code. Nothing left to do here.

Comment (rsimha-amp) [inline build-system/tasks/runtime-test.js]: Follow up fix removing `typeof` in #9769

**Pull request merged**